src/Components/Facility/Consultations/DailyRoundsFilter.tsx

```diff
@@ -33,14 +33,31 @@ export default function DailyRoundsFilter(props: Props) {
     errorClassName: "hidden",
   });
 
+  const countDefinedProperties = (obj) => {
+    return Object.values(obj).filter((val) => val !== undefined).length;
+  };
+
   return (
     <div className="flex flex-row-reverse items-center gap-4 md:flex-row">
       <Popover className="relative ">
         <Popover.Button>
-          <ButtonV2 variant="secondary" className="mr-5 border">
-            <CareIcon className="care-l-filter" />
-            {t("filter")}
-          </ButtonV2>
+          {filter == undefined ||
+          filter == null ||
+          countDefinedProperties(filter) == 0 ? (
+            <>
+              <ButtonV2 variant="secondary" className="mr-5 border">
+                <CareIcon className="care-l-filter" />
+                {t("filter")}
+              </ButtonV2>
+            </>
+          ) : (
+            <>
+              <ButtonV2 variant="primary" className="mr-5 border">
+                <CareIcon className="care-l-filter" />
+                {t("filter")}
+              </ButtonV2>
+            </>
+          )}
```

Comment:
This code is not readable. You are constructing two different buttons based on the condition, when only the `variant` prop is changing.

Instead pass the boolean condition to `variant` alone.


```suggestion
          <ButtonV2 variant={isFilterApplied ? "primary" : "secondary" className="mr-5 border">
            <CareIcon className="care-l-filter" />
            {t("filter")}
          </ButtonV2>
```

Reply:
Thanks for the review. Yes the code was not readable. I have made the suggested changes.